flink-runtime/src/main/java/org/apache/flink/streaming/api/datastream/ConnectedStreams.java

```diff
@@ -523,4 +529,14 @@ private <R> SingleOutputStreamOperator<R> doTransform(
 
         return returnStream;
     }
+
+    private boolean isEnableAsyncState() {
+        boolean enableAsyncState = false;
+        if ((inputStream1 instanceof KeyedStream) && (inputStream2 instanceof KeyedStream)) {
+            enableAsyncState =
+                    ((KeyedStream<?, ?>) inputStream1).isEnableAsyncState()
+                            && ((KeyedStream<?, ?>) inputStream2).isEnableAsyncState();
+        }
+        return enableAsyncState;
+    }
```

Comment:
I'd suggest we keep a member `boolean isEnableAsyncState`  here. Thus user could perform `enableAsyncState()` on `ConnectedStreams`, WDYT?

Reply:
👍, I added `enableAsyncState()` for `ConnectedStreams`.